generators/app/templates/ext-wizard/src/wizards/db-ops/wizard/pages/page1.ts

```diff
@@ -0,0 +1,151 @@
+    }
+
+    async start() : Promise<boolean> {
+        let descriptionComponent =
+                {component: this.view.modelBuilder.text().component(),
+                title: 'This Wizard will demo commonly-used connection operations '
+                + 'and UI elements. You will select one of your existing connections, and '
+                + 'then choose an operation to perform on that connection.'};
+        let serverComponent = await this.createServerDropdown();
+        let databaseComponent = await this.createDatabaseDropdown();
+
+        let layout = {componentWidth: this.width};
+        let formBuilder = this.view.modelBuilder.formContainer().withFormItems(
+            [descriptionComponent, serverComponent, databaseComponent], layout);
+
+        await this.view.initializeModel(formBuilder.component());
+        return true;
+    }
+
+	async onPageEnter(): Promise<boolean> {
+		let r1 = await this.populateServerDropdown();
+		let r2 = await this.populateDatabaseDropdown();
+		return r1 && r2;
+    }
+
+    private async createServerDropdown(): Promise<azdata.FormComponent> {
+		this.serverDropdown = this.view.modelBuilder.dropDown().withProperties({
+			required: true,
+		}).component();
+
+        // Handle server changes
+		this.serverDropdown.onValueChanged(async () => {
+            if (this.serverDropdown) {
+                let connection = this.serverOptions.get(String(this.serverDropdown.value));
+
+                if (connection) {
+                    this.model.server = connection;
+                    await this.populateDatabaseDropdown();
+                }
+            }
+		});
+
+		return {
+			component: this.serverDropdown,
+			title: 'Select a Server:'
+		};
+    }
+
+    private async createDatabaseDropdown(): Promise<azdata.FormComponent> {
+		this.databaseDropdown = this.view.modelBuilder.dropDown().withProperties({
+			required: true,
+		}).component();
+
+		// Handle database changes
+		this.databaseDropdown.onValueChanged(async () => {
+            if (this.databaseDropdown) {
+                this.model.database = String(this.databaseDropdown.value);
+            }
+		});
+
+		return {
+			component: this.databaseDropdown,
+			title: 'Select a Database:'
+		};
+    }
+
+    private async populateServerDropdown(): Promise<boolean> {
+        if (!this.serverDropdown) {
+            return false;
+        }
+		let connectionsExist = await this.getServerValues();
+		if (!connectionsExist) {
+            vscode.window.showErrorMessage("Connect to a server first to use this wizard.")
+			return false;
+		}
+
+        let connections : string[] = Array.from(this.serverOptions.keys());
+        let firstConnection = this.serverOptions.get(connections[0]);
+        if (firstConnection) {
+            this.model.server = firstConnection;
+            this.serverDropdown.updateProperties({
+                values: connections
+            });
+        }
+		return firstConnection !== undefined;
```

Comment:
please either use space or tabs.

Reply:
space is used in existing files

---
In reply to: [465866150](https://github.com/llali/generator-azuredatastudio/pull/25#discussion_r465866150) [](ancestors = 465866150)